subprojects/docs/src/docs/release/notes.md

```diff
@@ -138,9 +140,18 @@ These methods have been deprecated and the `create()` or `register()` methods sh
 
 ## Potential breaking changes
 
-### Kotlin DSL breakages
+### Changes to the Gradle Kotlin DSL
 
-- `project.java.sourceSets` is now `project.sourceSets`
+The Kotlin DSL now enables the _SAM conversion for Kotlin functions_ Kotlin compiler feature in order to expose an uniform `org.gradle.api.Action<T>` based API to both Groovy DSL and Kotlin DSL scripts.
```

Comment:
This reads a bit too technical IMO. Can we simplify the message here a bit? I fear throwing around technical terms a lot indicates that this is complex (unjustly).

Reply:
We could leave some of the details out.

> This Kotlin DSL release takes advantage of experimental Kotlin compiler features in order to...